File: back-end/mercado-inteligente-spring/src/main/java/br/edu/usf/mercadointeligentespring/core/usecase/CreateUser.java
package br.edu.usf.mercadointeligentespring.core.usecase;

import br.edu.usf.mercadointeligentespring.core.domain.User;
import br.edu.usf.mercadointeligentespring.core.mapper.UserMapper;
import br.edu.usf.mercadointeligentespring.gateways.UserGateway;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;


@Service
@RequiredArgsConstructor
public class CreateUser {

    private final UserGateway userGateway;

    private final UserMapper userMapper;

    public User execute(User user) {
        var userSaved = userGateway.create(userMapper.toDTO(user));
        return userMapper.toDomain(userSaved);
    }
}

File: back-end/mercado-inteligente/src/main/kotlin/br/edu/usf/core/usecase/CreateUser.kt
package br.edu.usf.core.usecase

import br.edu.usf.controllers.mapper.UserMapper
import br.edu.usf.core.domain.User
import br.edu.usf.gateways.UserGateway
import javax.enterprise.context.ApplicationScoped

@ApplicationScoped
class CreateUser(val userGateway: UserGateway, val userMapper: UserMapper) {
    fun execute(user: User): User {
        val userSaved = userGateway.create(userMapper.toDTO(user))
        return userMapper.toDomain(userSaved);
    }
}